src/main/java/hudson/remoting/Channel.java

```diff
@@ -1389,7 +1400,25 @@ public void dumpExportTable(PrintWriter w) throws IOException {
      * @see #lastHeard
      */
     public long getLastHeard() {
-        return lastHeard;
+        // TODO - this is not safe against clock skew and is called from jenkins core (and potentially plugins)
+        try {
+            lastHeardLock.readLock().lock();
+            return lastHeard;
+        }
+        finally {
+            lastHeardLock.readLock().unlock();
+        }
+    }
+
+    private void updateLastHeard() {
+        try {
+            lastHeardLock.writeLock().lock();
+            // TODO - this is not safe against clock skew and is called from jenkins core (and potentially plugins)
+            lastHeard = System.currentTimeMillis();
```

Comment:
(Which actually __is__ protected against by using a volatile long - Every day is a school day).